cloudvolume/datasource/precomputed/mesh/sharded.py

```diff
@@ -0,0 +1,199 @@
+        self.meta = meta
+        self.cache = cache
+        self.config = config
+        self.readonly = bool(readonly)
+
+        spec = ShardingSpecification.from_dict(self.meta.info['sharding'])
+        self.reader = ShardReader(meta, cache, spec)
+
+        self.vertex_quantization_bits = self.meta.info['vertex_quantization_bits']
+        self.lod_scale_multiplier = self.meta.info['lod_scale_multiplier']
+        self.transform = np.array(self.meta.info['transform'] + [0,0,0,1]).reshape(4,4)
+    
+        if np.any(self.transform * np.array([[0,1,1,1],[1,0,1,1],[1,1,0,1],[1,1,1,0]])):
+            raise ValueError(red("Non-scale homogeneous transforms are not implemented"))
+
+    @property
+    def path(self):
+        return self.meta.mesh_path
+    
+    def get(self, segids, lods=None):
+        """Fetch meshes at all levels of details.
+
+        Parameters:
+        segids : int or [int ..]
+            Scalar or list of integer segment IDs.
+        lods : int, [int, ..] or None
+            Level of detail(s) to retrieve.  0 is highest level of detail.
+            None will fetch meshes at all levels.
+
+        Returns:
+        [Mesh, ..] or [[Mesh, ..], ..]
+            A list of meshes for each ssegment ID is returned.
+            An additional layer is added when multiple lods are requested.
+
+        Reference:
+            https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md
+        """
+        seg_list_return = True
+        if type(segids) in (int, float):
+            seg_list_return = False
+            segids = [ int(segids) ]
+
+        results = []
+        for segid in segids:
+            # Read the manifest (with a tweak to sharding.py to get the offset)
+            binary, shard_file_offset = self.reader.get_data(segid, self.meta.mesh_path, return_offset=True)
+            manifest = MultiLevelPrecomputedMeshManifest(binary)
+
+            single_lod_return = False
+            if isinstance(lods, int):
+                single_lod_return = True
+                lods = [lods]
+            elif lods == None:
+                lods = list(range(manifest.num_lods))
+
+            if any(lod >= manifest.num_lods for lod in lods):
+                raise ValueError(red("LOD value out of range (%d > %d)" % (max(lods), manifest.num_lods)))
+
+            # Read the data for all LODs
+            fragment_sizes = [ np.sum(lod_fragment_sizes) for lod_fragment_sizes in manifest.fragment_offsets ]
+            total_fragment_size = np.sum(fragment_sizes)
+
+            # Kludge to hijack sharding.py to read the data
+            shard_file_name = self.reader.get_filename(segid)
+            full_path = self.reader.meta.join(self.reader.meta.cloudpath, self.path)
+            stor =  SimpleStorage(full_path)
+
+            meshes = []
+            for lod in lods:
+                lod_binary = stor.get_file(shard_file_name,
```

Comment:
Ideally, you could use `Storage.get_files` to extract all the LOD binaries at once so you don't take the performance hit. Otherwise, it's conceivably faster to do what you were doing before and just discard the extra LODs. I think it's also fine to not worry about performance right now and just get something working out. I'll have to tune this for performance later anyway.

Reply:
> My feeling is that retrieving all the LODs all the time is kind of a waste of the benefits of the format.

... is what is being implemented here. Maybe a minor optimization would be to request adjacent LODs in a single request. Still, this change is good change for any request that does not include the full resolution mesh. (Note that all of the fragments for a given lod are retrieved in this call.)